src/plugins/intel_cpu/src/nodes/scaled_attn.cpp

```diff
@@ -749,10 +635,44 @@ void ScaledDotProductAttention::initSupportedPrimitiveDescriptors() {
     if (!supportedPrimitiveDescriptors.empty())
         return;
     auto rtPrecision = getOriginalInputPrecisionAtPort(0);
+    auto orginSDPInputNumber = getOriginalInputsNumber() - (m_config.config.fuse_concat ? 2 : 0);
+
+    auto fp16_kvcache_applicable = [&]() -> bool {
+        bool enable_fp16_kvcache = true;
+        const char* disable = std::getenv("OV_DISABLE_SDPA_KVCACHE_FP16");
+        if (disable && std::atoi(disable) > 0) {
+            enable_fp16_kvcache = false;
+        }
+        if (!enable_fp16_kvcache) return false;
+
+        if (!m_config.config.fuse_concat || !mayiuse(cpu_isa_t::avx2) || rtPrecision== ov::element::bf16) return false;
+
+        // applicable only in case of stateful model.
```

Comment:
Do we really need so complex condition? Cannot we just save information that original op type is ScaledDotProductAttentionWithKVCache?

Reply:
Simplified.